generic3g/MAPL_Generic.F90

```diff
@@ -143,10 +153,23 @@ module mapl3g_Generic
       procedure :: gridcomp_connect_all
    end interface MAPL_ConnectAll
 
-
+   ! MAPL_ResourceGet
+   !  This will have 4 specific procedures:
+   !     scalar value from hconfig
+   !     scalar value from metacomp
```

Comment:
We may need 2 cases, only: scalar and array/sequence. Both would be for _gridcomp_. If we want to be able to call `MAPL_ResourceGet` for **both** _gridcomp_ **and** _metacomp_, it would double, obviously. If we want to be able call `MAPL_ResourceGet` for _hconfig_, as well, there would be 2 more. I don't think we really need the version for the _hconfig_ itself, since this is in `MAPL_Generic.F90` in `generic3g`. The _hconfig_ is not tied to anything else in MAPL (except exception handling). _hconfig_utils_ has a public procedure that can be used if we want to get an _hconfig_ value elsewhere. So my current plan is to fold the _hconfig_ version into the _gridcomp_ version. We should talk.

Reply:
Now there are two specific procedures for `MAPL_ResourceGet`:

1. with gridcomp (calls 2.)
2. with hconfig

The hconfig is accessible with `MAPL_GridCompGet`, so an intermediate procedure is not required for the gridcomp. However, if there is a use case for a standalone metacomp procedure, we can add that. Of course this will double for arrays/sequences.